### PR TITLE
[SPARK-31529][SQL] Remove extra whitespaces in formatted explain

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Remove all the extra whitespaces in the formatted explain.

### Why are the changes needed?

The number of extra whitespaces of the formatted explain becomes different between master and branch-3.0. This causes a problem that whenever we backport formatted explain related tests from master to branch-3.0, it will fail branch-3.0. Besides, extra whitespaces are always disallowed in Spark. Thus, we should remove them as possible as we can.

### Does this PR introduce any user-facing change?

No, formatted explain is newly added in Spark 3.0.

### How was this patch tested?

Updated sql query tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala [hunk 1]
```diff
@@ -195,24 +195,27 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]] extends TreeNode[PlanT
   }
 
   def verboseStringWithOperatorId(): String = {
-    val codegenIdStr =
-      getTagValue(QueryPlan.CODEGEN_ID_TAG).map(id => s"[codegen id : $id]").getOrElse("")
-    val operatorId = getTagValue(QueryPlan.OP_ID_TAG).map(id => s"$id").getOrElse("unknown")
-    val baseStr = s"($operatorId) $nodeName $codegenIdStr"
```
Comment (Ngone51) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala:201]: There will be an extra space behind `$nodeName` when `$codegenIdStr` is empty.

**Comment by Ngone51** [comment]
This is root cause of #28303. cc @cloud-fan @dongjoon-hyun @HyukjinKwon 

also cc @Eric5553

**Review by cloud-fan — APPROVED** [review]

**Comment by Ngone51** [comment]
retest this please.

**Comment by dongjoon-hyun** [comment]
This sounds like a follow-up. Do you know which commit make this difference between master and 3.0?

**Comment by Ngone51** [comment]
@dongjoon-hyun this one: https://github.com/apache/spark/commit/eba2076ca32b0b497c1f65ad9d7c6297591c7e9e

**Comment by cloud-fan** [comment]
retest this please

**Comment by Ngone51** [comment]
retest this please.

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by cloud-fan** [comment]
@Ngone51 can you send a PR for 3.0 if there is something we need to fix?

**Comment by Ngone51** [comment]
Sure, thanks!

**Pull request closed without merging**